analytics/track.go

```diff
@@ -0,0 +1,80 @@
+package analytics
+
+import (
+	"bytes"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+const poolSize = 10
+const bufferSize = 100
+
+// Tracker ...
+type Tracker interface {
+	Enqueue(eventName string, properties map[string]interface{})
+	Wait()
+}
+
+type tracker struct {
+	jobs             chan *bytes.Buffer
+	waitGroup        *sync.WaitGroup
+	client           Client
+	sharedProperties map[string]interface{}
+}
+
+// NewDefaultTracker ...
+func NewDefaultTracker(shared map[string]interface{}) Tracker {
+	return NewTracker(make(chan *bytes.Buffer, bufferSize), &sync.WaitGroup{}, poolSize, NewDefaultClient(), shared)
+}
+
+// NewTracker ...
+func NewTracker(jobs chan *bytes.Buffer, waitGroup *sync.WaitGroup, size int, client Client, sharedProperties map[string]interface{}) Tracker {
```

Comment:
I feel like it's not a good idea to let the caller decide the "pool size". Is this really expected that steps provide different size values? Why don't we just define a reasonable default value here, and later add a new constructor with a size parameter if needed?

Reply:
there is a default constructor that pins down that as well.